main.py

```diff
@@ -0,0 +1,148 @@
+import csv
+import logging
+
+
+class Account:
+    def __init__(self, name):
+        self.name = name
+        self.balance = 0
+        self.transactions = []
+
+    def add_money(self, amount):
+        self.balance += amount
+
+    def substract_money(self, amount):
+        self.balance -= amount
+
+    def add_transaction(self, date, narrative, amount, paid): # if paid is True -> person sent amount
+                                                               # if paid is False -> person received amount
+        self.transactions.append((date, narrative, amount, paid))
+        self.transactions.sort()
+
+    def print_transactions(self):
+        for transaction in self.transactions:
+            if transaction[3]:
+                print(self.name + " sent amount " + str(transaction[2]) + " on " + transaction[0] + "\nnarrative : " + transaction[1] + "\n")
+            else:
+                print(self.name + " received amount " + str(transaction[2]) + " on " + transaction[0] + "\nnarrative : " + transaction[1] + "\n")
+
+    def __str__(self):
+        if self.balance >= 0:
+            return "Name: " + self.name + "\nhas money : " + str(self.balance) + "\n"
+        else:
+            return "Name: " + self.name + "\nhas debt of : " + str(abs(self.balance)) + "\n"
+
+# Program start
+
+# Add logging
+logger = logging.getLogger(__name__)
+logging.basicConfig(filename='SupportBank.log', filemode='w', level=logging.DEBUG)
+
+logger.info("Starting to run SupportBank")
+
+accounts = []
+
+
+def find_account(acc_name):
+    for acc in accounts:
+        if acc.name == acc_name:
+            return acc
+    return False
+
+
+def correct_date(date):
+    date_values = date.split('/')
+
+    # Check if the values are int type
+    try:
+        day = int(date_values[0])
+        month = int(date_values[1])
+        year = int(date_values[2])
+    except ValueError:
+        return False
+
+    # Check if the values are physically possible
+    if day < 1 or day > 31 or month < 1 or month > 12:
+        return False
+
+    return True
+
+
+
+csvFileName = input("Enter the name of the csv file: ")
+
+# Read from csv file
+logger.info("Reading from " + csvFileName + " file")
+
+with open(csvFileName, newline='') as csvfile:
+    csvreader = csv.reader(csvfile, delimiter=',', quotechar='|')
+    csvlist = list(csvreader)
+
+    first = True
+
+    for row in csvlist:
+        # Skip the first line in the CSV
+        if first is True:
+            first = False
+            continue
+
+        # Substract money from sender
+
+        # Check if the amount is a float
+        try:
+            float(row[4])

```

Comment:
Similar to the comment above, we should try to avoid using `row[4]` as it is not very descriptive and somebody with less context would struggle to understand what that is.